main.py

```diff
@@ -80,11 +80,13 @@ def getWriteLock(self, T: Transaction, x: str) -> tuple[str, Transaction]:
         
 
         if len(self.table[x]) == 0:
-            self.table[x].append(self.LockTuple(T, "W", State.GRANTED))
-            return "W", None
+            newLock = self.LockTuple(T, "W", State.GRANTED)
```

Comment:
Just extracting the newLock variable for better readability, the behavior should be the same